python/ray/tune/ray_trial_executor.py

```diff
@@ -68,8 +68,9 @@ def __init__(self):
 
     def get(self, trainable_cls):
         """Gets the wrapped trainable_cls, otherwise calls ray.remote."""
+        runtime_env = {"env_vars": {"TUNE_ORIG_WORKING_DIR": os.getcwd()}}
         if trainable_cls not in self._cache:
-            remote_cls = ray.remote(trainable_cls)
+            remote_cls = ray.remote(runtime_env=runtime_env)(trainable_cls)
```

Comment:
Doesn't this override the user-specified runtime env? Maybe we can do it with `tune.orig_working_dir()` instead? (and an attribute for Class API)

Reply:
I thought env variables are "merged" rather than "overridden". See docs [here](https://docs.ray.io/en/master/ray-core/handling-dependencies.html#runtime-environments).

Nonetheless, are you proposing that for function trainables, we expose
`with tune.orig_working_dir()` and with class trainables, we expose `with self.orig_working_dir()`?
I was thinking env variables can be more light. In my mind, I am less convinced that this should be an "offical" API as it's easy for one to shoot their foot. This is just a workaround for those who are very clear about what they are doing.

Reply:
Yeah, I wasn't sure myself how it worked, thanks for clarifying!

I don't really like the concept of using env vars inside trainables for this, even if it's an advanced API. I'd rather make it explicit in documentation. That way we avoid doing anything with runtime envs, too (even if it's not an issue).

Reply:
Ok, good points. Let's go ahead with this